classes/submission/SubmissionFileDAODelegate.inc.php

```diff
@@ -134,11 +138,36 @@ function insertObject($submissionFile, $sourceFile, $isUpload = false) {
 	 * @return boolean
 	 */
 	function updateObject($submissionFile, $previousFile) {
+		// MS SQL doesn't accept the UPDATE on a primary key IDENTITY
+		$isSqlServer = Config::getVar('database', 'ms_sql');
+
+		$params = array((int)$submissionFile->getRevision(),
```

Comment:
So the issue is updating the file ID or revision, right? I don't *think* OJS ever changes these values for existing entries, so it should be possible to remove them from the `UPDATE` entirely for all DB types. We would of course have to test for this!

Reply:
I removed the file ID for the update section, but I can't make the same change for the insertion as the table is composed with a two columns primary key.

We need to insert a new revision each time in the table and SQL server doesn't really like it.

To make it functionnal within all databases without the use of "IDENTITY_INSERT ON/OFF", a refactor is needed to split the table in 2 (one with the file ID and revision ID as a foreign key and another one with the revision ID + associated data).